src/viewPanel/sessionWidget.ts

```diff
@@ -150,23 +150,29 @@ export class SessionWidget extends BoxPanel {
   private _onTabsChanged(): void {
     Object.entries(this._model.tabs).forEach(
       async ([tabName, tabData], idx) => {
-        const model = new TabModel({
-          tabName,
-          tabData,
-          model: this._model,
-          rendermime: this._rendermime,
-          context: this._context,
-          notebookTracker: this._notebookTracker,
-          dataLoaded: this._dataLoaded
-        });
-        const tabWidget = new TabView({ model });
-
-        this._tabPanel.addTab(tabWidget, idx + 1);
+        if (tabName in this._tabsModels) {
+          this._tabsModels[tabName].tabData = tabData;
+        } else {
```

Comment:
@trungleduc @hbcarlos I suppose we'll need to be granular on the changes here?

We don't want to re-render all viewers (plots) every time there is a change in the tabs data. Instead, we want to:
- create new viewers if there are new ones (what my `if` condition is doing here)
- delete old viewers if they are not part of the shared model anymore
- update viewers states if they need updating

How do you see the implementation for this? Should we do it all manual? I suppose the YJS event should contain everything about what actually happens in the tabs change?

Reply:
> I suppose the YJS event should contain everything about what actually happens in the tabs change?

I think it should be if the tab key is deep-observed

Reply:
Agree. We should not re-render everything all the time.

> I suppose the YJS event should contain everything about what actually happens in the tabs change?

Yes. Each tab should be a map, and then we can use deepobserve to listen for changes on each tab.